src/adagrams.js

```diff
@@ -1,15 +1,160 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+  };
+
+  const handLetters = [];
+  const letters = Object.keys(letterPool);
+
+  while (handLetters.length < 10) {
+    let letterIndex = Math.floor((Math.random() * letters.length));
+    let letter = letters[letterIndex];
+    
+    if (letterPool[letter] > 0) {
+      letterPool[letter]--;
+      handLetters.push(letter);
+    }
+  }
+  return handLetters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const lettersInHandCopy = [...lettersInHand];
+
+  for (let letter of input.toUpperCase()) {
```

Comment:
Since the `letter` variable isn't reassigned, you should use `const` over `let`.